esmvalcore/_data_finder.py

```diff
@@ -67,16 +69,35 @@ def get_start_end_year(filename):
         filename_list[ind] for ind, _ in enumerate(pos_ydates)
         if pos_ydates_r[ind] or pos_ydates_l[ind]
     ]
-
+    start_year = None
+    end_year = None
     if len(dates) == 1:
         start_year = int(dates[0][:4])
         end_year = start_year
     elif len(dates) == 2:
         start_year, end_year = int(dates[0][:4]), int(dates[1][:4])
     else:
-        raise ValueError('Name {0} dates do not match a recognized '
-                         'pattern'.format(name))
-
+        # Slower than just parsing the name
```

Comment:
Is this needed? The preprocessor function `extract_time` will already take care of removing unnecessary data.

Reply:
Yes, but extract time is performed after several steps, including concatenation. If we don limit the reading to the really required files we can experiment errors before `extract_time` due to problems in files that we don't need at all

Reply:
If you expect such errors, I guess this is OK